beacon-chain/rpc/eth/node/handlers.go

```diff
@@ -32,3 +56,363 @@ func (s *Server) GetSyncStatus(w http.ResponseWriter, r *http.Request) {
+		Attnets:   hexutil.Encode(s.MetadataProvider.Metadata().AttnetsBitfield()),
+	}
+
+	resp := &GetIdentityResponse{
+		Data: &Identity{
+			PeerId:             peerId,
+			Enr:                enr,
+			P2PAddresses:       p2pAddresses,
+			DiscoveryAddresses: discoveryAddresses,
+			Metadata:           meta,
+		},
+	}
+	http2.WriteJson(w, resp)
+}
+
+// GetPeer retrieves data about the given peer.
+func (s *Server) GetPeer(w http.ResponseWriter, r *http.Request) {
+	_, span := trace.StartSpan(r.Context(), "node.GetPeer")
+	defer span.End()
+
+	rawId := mux.Vars(r)["peer_id"]
+	if rawId == "" {
+		http2.HandleError(w, "peer_id is required in URL params", http.StatusBadRequest)
+		return
+	}
+
+	peerStatus := s.PeersFetcher.Peers()
+	id, err := peer.Decode(rawId)
+	if err != nil {
+		http2.HandleError(w, "Invalid peer ID: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+	enr, err := peerStatus.ENR(id)
+	if err != nil {
+		if errors.Is(err, peerdata.ErrPeerUnknown) {
+			http2.HandleError(w, "Peer not found: "+err.Error(), http.StatusNotFound)
+			return
+		}
+		http2.HandleError(w, "Could not obtain ENR: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	serializedEnr, err := p2p.SerializeENR(enr)
+	if err != nil {
+		http2.HandleError(w, "Could not obtain ENR: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	p2pAddress, err := peerStatus.Address(id)
+	if err != nil {
+		http2.HandleError(w, "Could not obtain address: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	state, err := peerStatus.ConnectionState(id)
+	if err != nil {
+		http2.HandleError(w, "Could not obtain connection state: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	direction, err := peerStatus.Direction(id)
+	if err != nil {
+		http2.HandleError(w, "Could not obtain direction: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if eth.PeerDirection(direction) == eth.PeerDirection_UNKNOWN {
+		http2.HandleError(w, "Peer not found", http.StatusNotFound)
+		return
+	}
+
+	v1ConnState := migration.V1Alpha1ConnectionStateToV1(eth.ConnectionState(state))
+	v1PeerDirection, err := migration.V1Alpha1PeerDirectionToV1(eth.PeerDirection(direction))
+	if err != nil {
+		http2.HandleError(w, "Could not handle peer direction: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	resp := &GetPeerResponse{
+		Data: &Peer{
+			PeerId:             rawId,
+			Enr:                "enr:" + serializedEnr,
+			LastSeenP2PAddress: p2pAddress.String(),
+			State:              strings.ToLower(v1ConnState.String()),
+			Direction:          strings.ToLower(v1PeerDirection.String()),
+		},
+	}
+	http2.WriteJson(w, resp)
+}
+
+// GetPeers retrieves data about the node's network peers.
+func (s *Server) GetPeers(w http.ResponseWriter, r *http.Request) {
+	_, span := trace.StartSpan(r.Context(), "node.GetPeers")
+	defer span.End()
+
+	states := r.URL.Query()["state"]
```

Comment:
more of a question but what does this returned data look like? it's a string array of states? would be nice if a comment can be added.

Reply:
I don't think we need a comment when you can hover over the variable to see the return type